src/service/resources-impl.js

```diff
@@ -435,6 +435,69 @@ export class Resources {
   }
 
   /**
+   * Runs the specified subtree "show" mutation and ensures that measures
+   * and layouts performed for the affected elements.
+   * @param {!Element} root
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  showSubtree(root, mutator) {
+    // This is a two-vsync operation: we can only measure after the mutation
+    // has been done.
+    return new Promise(resolve => {
+      this.vsync_.mutate(() => {
+        mutator();
+        this.vsync_.measure(() => {
+          const newTop = this.viewport_.getLayoutRect(root).top;
+          this.setRelayoutTop_(newTop);
+
+          // Remeasure the previously hidden elements.
+          const ampElements = root.getElementsByClassName('-amp-element');
+          for (let i = 0; i < ampElements.length; i++) {
+            const r = this.getResourceForElement(ampElements[i]);
+            r.requestMeasure();
+          }
+
+          this.schedulePass(100);
+          resolve();
+        });
+      });
+    });
+  }
+
+  /**
+   * Runs the specified subtree "hide" mutation and ensures that the affected
+   * elements are unloaded and other elements are remeasured.
+   * @param {!Element} root
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  hideSubtree(root, mutator) {
+    return new Promise(resolve => {
+      this.vsync_.run({
+        measure: state => {
+          state.oldTop = this.viewport_.getLayoutRect(root).top;
+        },
+        mutate: state => {
+          mutator();
+          this.setRelayoutTop_(state.oldTop);
+
+          // Unload AMP elements within.
+          const ampElements = root.getElementsByClassName('-amp-element');
```

Comment:
That's triggered automatically via `setRelayoutTop_` above. The elements within this root do not get triggered specifically because they never been laid out before - so it doesn't try until some better signal.